components/Deck/TreePanel/TreeNode.js

```diff
@@ -135,13 +135,13 @@ class TreeNode extends React.Component {
         const {isDragging, connectDragSource, nodeIndex} = this.props;
         //adapt URLs based on the current page
         let nodeSelector = {
-            id: `${this.props.rootNode.id}/${this.props.rootNode.slug}`,
+            id: this.props.rootNode.id,
             stype: this.props.item.get('type'),
             sid: this.props.item.get('id'),
             spath: this.props.item.get('path')
         };
         let canEdit = !this.props.permissions.readOnly && this.props.permissions.edit && this.props.ContentStore.mode !== 'edit' && !this.props.showThumbnails;
-        let nodeURL = TreeUtil.makeNodeURL(nodeSelector, this.props.page, 'view');
+        let nodeURL = TreeUtil.makeNodeURL(nodeSelector, this.props.page, 'view', this.props.rootNode.slug);
```

Comment:
This is the only code line where slug should be forwarded?

Reply:
No, there are lots of places we still need to add it, but for now I just put it in a few and will see how we manage that later. I'm not sure it's the best idea to do it like this, and would like to revisit later. This PR is a bugfix, not a feature PR, let's focus on that :)

Reply:
Thanks!